lib/algo/ParametricDMDc.h

```diff
@@ -0,0 +1,194 @@
+                       std::string rbf = "G",
+                       std::string interp_method = "LS",
+                       double closest_rbf_val = 0.9,
+                       bool reorthogonalize_W = false)
+{
+    CAROM_VERIFY(parameter_points.size() == dmdcs.size());
+    CAROM_VERIFY(dmdcs.size() > 1);
+    for (int i = 0; i < dmdcs.size() - 1; i++)
+    {
+        CAROM_VERIFY(dmdcs[i]->d_dt == dmdcs[i + 1]->d_dt);
+        CAROM_VERIFY(dmdcs[i]->d_k == dmdcs[i + 1]->d_k);
+    }
+    CAROM_VERIFY(closest_rbf_val >= 0.0 && closest_rbf_val <= 1.0);
+
+    int mpi_init, rank;
+    MPI_Initialized(&mpi_init);
+    if (mpi_init == 0) {
+        MPI_Init(nullptr, nullptr);
+    }
+
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+
+    std::vector<CAROM::Matrix*> bases;
+    std::vector<CAROM::Matrix*> A_tildes;
+    std::vector<CAROM::Matrix*> B_tildes;
+    for (int i = 0; i < dmdcs.size(); i++)
+    {
+        bases.push_back(dmdcs[i]->d_basis);
+        A_tildes.push_back(dmdcs[i]->d_A_tilde);
+        B_tildes.push_back(dmdcs[i]->d_B_tilde);
+    }
+
+    int ref_point = getClosestPoint(parameter_points, desired_point);
+    std::vector<CAROM::Matrix*> rotation_matrices = obtainRotationMatrices(
+                parameter_points,
+                bases, ref_point);
+
+    CAROM::MatrixInterpolator basis_interpolator(parameter_points,
+            rotation_matrices, bases, ref_point, "B", rbf, interp_method, closest_rbf_val);
+    CAROM::Matrix* W = basis_interpolator.interpolate(desired_point,
+                       reorthogonalize_W);
+
+    CAROM::MatrixInterpolator A_tilde_interpolator(parameter_points,
+            rotation_matrices, A_tildes, ref_point, "R", rbf, interp_method,
+            closest_rbf_val);
+    CAROM::Matrix* A_tilde = A_tilde_interpolator.interpolate(desired_point);
+
+    CAROM::MatrixInterpolator B_tilde_interpolator(parameter_points,
+            rotation_matrices, B_tildes, ref_point, "R", rbf, interp_method,
+            closest_rbf_val);
+    CAROM::Matrix* B_tilde = B_tilde_interpolator.interpolate(desired_point);
+
+    CAROM::MatrixInterpolator control_interpolator(parameter_points,
+            rotation_matrices, controls, ref_point, "R", rbf, interp_method,
+            closest_rbf_val);
+
+    controls_interpolated = control_interpolator.interpolate(desired_point);
+
+    // Calculate the right eigenvalues/eigenvectors of A_tilde
+    ComplexEigenPair eigenpair = NonSymmetricRightEigenSolve(A_tilde);
+    std::vector<std::complex<double>> eigs = eigenpair.eigs;
+
+    // Calculate phi (phi = W * eigenvectors)
+    CAROM::Matrix* phi_real = W->mult(eigenpair.ev_real);
+    CAROM::Matrix* phi_imaginary = W->mult(eigenpair.ev_imaginary);
+
+    parametric_dmdc = new T(eigs, phi_real, phi_imaginary, B_tilde,
+                            dmdcs[0]->d_k,dmdcs[0]->d_dt,
+                            dmdcs[0]->d_t_offset, dmdcs[0]->d_state_offset, W);
+
+//    delete W;
```

Comment:
Decide whether these pointers should be deleted?

Reply:
I keep both of them. Added a commit to delete the commented lines